test/battle/move_effect/instruct.c

```diff
@@ -0,0 +1,217 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gMovesInfo[MOVE_INSTRUCT].effect == EFFECT_INSTRUCT);
+}
+
+DOUBLE_BATTLE_TEST("Instruct fails if target hasn't made a move")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WOBBUFFET) { Moves(MOVE_TACKLE, MOVE_POUND, MOVE_SCRATCH, MOVE_CELEBRATE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(playerLeft, MOVE_INSTRUCT, target: playerRight); MOVE(playerRight, MOVE_TACKLE, target: opponentLeft); }
+    } SCENE {
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_INSTRUCT, playerRight);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerRight);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Instruct fails if move is banned by Instruct")
+{
+    ASSUME(gMovesInfo[MOVE_BIDE].instructBanned == TRUE);
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WOBBUFFET) { Moves(MOVE_TACKLE, MOVE_POUND, MOVE_SCRATCH, MOVE_BIDE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(playerRight, MOVE_BIDE); MOVE(playerLeft, MOVE_INSTRUCT, target: playerRight); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_BIDE, playerRight);
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_INSTRUCT, playerRight);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Instruct-called move targets the target of the move picked on its last use")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WOBBUFFET) { Moves(MOVE_TACKLE, MOVE_POUND, MOVE_SCRATCH, MOVE_FAKE_OUT); }
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(playerRight, MOVE_TACKLE, target: opponentLeft); MOVE(playerLeft, MOVE_INSTRUCT, target: playerRight); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerRight);
+        HP_BAR(opponentLeft);
+        NOT HP_BAR(opponentRight);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_INSTRUCT, playerRight);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, playerRight);
```

Comment:
I might be misunderstanding instruct but is it coming from `playerRight` and not `playerLeft` as specified in `TURN`?

Reply:
That's because `tryinstruct` occurs before the animation and changes the attacker and target to be that of the move called. I can make it so it actually comes from the one using the move if that's better

Reply:
Do you know how this is handled in vanilla?

Reply:
No I don't. My change was just gonna be: save the attacker, change the attacker to be the user of Instruct, play the move's animation, restore the attacker

Reply:
ye, then I would do the change since it is simple enough and avoids confusion.

Reply:
ye seems logical